relayer/cmd/subscribe_beefy.go

```diff
@@ -0,0 +1,204 @@
+			header, err := GetParaheads(nextBlockHash, relaychainConn)
+			if err != nil {
+				log.WithError(err).Error("Failed to get para heads")
+				return err
+			}
+			GetParaHeadData(header, parachainConn)
+		}
+	}
+}
+
+func GetParaheads(blockHash types.Hash, relaychainConn *relaychain.Connection) (string, error) {
+	type ParaId types.U32
+
+	type ParaHeadsQuery struct {
+		ParaId ParaId
+	}
+
+	query := ParaHeadsQuery{ParaId: 200}
+
+	encoded, err := types.EncodeToBytes(query)
+	if err != nil {
+		log.WithError(err).Error("Error")
+	}
+
+	allParaHeadsStorageKey, err := types.CreateStorageKey(
+		relaychainConn.GetMetadata(),
+		"Paras",
+		"Heads", encoded, nil)
+	if err != nil {
+		log.WithError(err).Error("Failed to create parachain header storage key")
+	}
+
+	// TODO1 - in polkadotjs webapp it seems possible to query this state via paras.heads without specifying any option
+	// and get all parachain heads as a response. eg: trying it on Rococo returns a lot of paraheads.
+	// I think we'll need this (to be able to check against the parachain headers hash
+	// of all headers in MMRLeaf.ParachainHeads), so we should get this working too.
+	response, err := relaychainConn.GetAPI().RPC.State.GetStorageRaw(allParaHeadsStorageKey, blockHash)
+	if err != nil {
+		log.WithError(err).Error("Failed to get parachain header")
+	}
+
+	// TODO2 - the above query returns some extra bytes, related the the HeadData type (try this state query in polkadotjs
+	// webapp for example). These extra bytes I think are the parachain ID, so the response type needs to account for
+	// this properly. the below is just a hack to get the actual header out. It's also not clear to me if the response
+	// contains the entire header, or just a hash of the header, or some truncated header? If it's the entire header,
+	// then great we can use it entirely instead of querying for it in a follow up call
+	header := response.Hex()
+	actualHeader := fmt.Sprintf("%s%s", "0x", header[6:70])
+	log.WithField("actualHeader", actualHeader).Info("Sliced header response into actual header")
+
+	log.WithField("header", actualHeader).Info("Got parachain header")
+	return actualHeader, nil
+}
+
+func GetParaHeadData(header string, parachainConn *parachain.Connection) {
+	headerHash, err := types.NewHashFromHexString(header)
+	if err != nil {
+		log.WithError(err).Error("Failed to create header hash")
+	}
+
+	log.WithFields(logrus.Fields{
+		"hash": headerHash.Hex(),
+	}).Info("Querying for parachain header")
+	headerData, err := parachainConn.Api().RPC.Chain.GetHeader(headerHash)
+	if err != nil {
+		log.WithError(err).Error("Failed to get parachain header")
+		// TODO3: This usually errors with "json: cannot unmarshal string into Go struct field Header.number of type uint32"
+		// Seems maybe related to the block number encoding thing on parachains vs relay chains, or maybe because our parachain
+		// header has a different type to the relay chain header and we need to setup that type?
```

Comment:
Fixed in bb013dcf16863df9bb8aedf0ce7779c30f1a654f , reversion of block number type did not work.